packages/server/src/threads/automation.ts

```diff
@@ -120,7 +120,7 @@ class Orchestrator {
   }
 
   async getMetadata(): Promise<AutomationMetadata> {
-    const metadataId = generateAutomationMetadataID(this._automation._id)
+    const metadataId = generateAutomationMetadataID(this._automation._id!)
```

Comment:
Not feedback just a thought - The constant usage of `!` on doc ids make me feel like we've done something wrong. The id should always be available, it's only in the cases of initial creation where the doc fields may be undefined. 

The correct thing is probably for us to make the `Document` fields non-optional to start using a different pattern like:

```ts
interface CreateCar {
  ...
}

interface Car extends CreateCar, Document {
  ...
}
```

Reply:
Yeah, I'd been thinking that as well - we should probably have a think about refactoring this.